### ISSUE TITLE
Angular 6 & typescript paths no longer working

### ISSUE THREAD

**Issue by vespertilian** [wallabyjs/public#1644]
### Issue description or question

After updating to angular 6 I am struggling to get the typescript paths to work with wallaby, they did work previously, with v5, so it's confounding me. 

I setup a basic example of my issue based on your Webpack sample 
https://github.com/vespertilian/ngCliWebpackSample

The tests work when I run ng test, just not with wallaby 

### Wallaby.js configuration file from sample project

```javascript
var wallabyWebpack = require('wallaby-webpack');
var path = require('path');

var compilerOptions = Object.assign(
  require('./tsconfig.json').compilerOptions,
  require('./src/tsconfig.spec.json').compilerOptions);

compilerOptions.module = 'CommonJs';

module.exports = function (wallaby) {

  var webpackPostprocessor = wallabyWebpack({
    entryPatterns: [
      'src/wallabyTest.js',
      'src/**/*spec.js'
    ],

    module: {
      rules: [
        {test: /\.css$/, loader: ['raw-loader']},
        {test: /\.html$/, loader: 'raw-loader'},
        {test: /\.ts$/, loader: '@ngtools/webpack', include: /node_modules/, query: {tsConfigPath: 'tsconfig.json'}},
        {test: /\.js$/, loader: 'angular2-template-loader', exclude: /node_modules/},
        {test: /\.json$/, loader: 'json-loader'},
        {test: /\.styl$/, loaders: ['raw-loader', 'stylus-loader']},
        {test: /\.less$/, loaders: ['raw-loader', 'less-loader']},
        {test: /\.scss$|\.sass$/, loaders: ['raw-loader', 'sass-loader']},
        {test: /\.(jpg|png)$/, loader: 'url-loader?limit=128000'}
      ]
    },

    resolve: {
      extensions: ['.js', '.ts'],
      modules: [
        path.join(wallaby.projectCacheDir, 'src/app'),
        path.join(wallaby.projectCacheDir, 'src'),
        'node_modules',
        path.join(__dirname, './typings')
      ],
      alias: {
        "@gtypes/*": path.join(__dirname, 'typings')
      }
    },
    node: {
      fs: 'empty',
      net: 'empty',
      tls: 'empty',
      dns: 'empty'
    }
  });

  return {
    files: [
      {pattern: 'src/**/*.+(ts|css|less|scss|sass|styl|html|json|svg)', load: false},
      {pattern: 'src/**/*.d.ts', ignore: true},
      {pattern: 'src/**/*spec.ts', ignore: true}
    ],

    tests: [
      {pattern: 'src/**/*spec.ts', load: false},
      {pattern: 'src/**/*e2e-spec.ts', ignore: true}
    ],

    testFramework: 'jasmine',

    compilers: {
      '**/*.ts': wallaby.compilers.typeScript(compilerOptions)
    },

    middleware: function (app, express) {
      var path = require('path');
      app.use('/favicon.ico', express.static(path.join(__dirname, 'src/favicon.ico')));
      app.use('/assets', express.static(path.join(__dirname, 'src/assets')));
    },

    env: {
      kind: 'electron'
    },

    postprocessor: webpackPostprocessor,

    setup: function () {
      window.__moduleBundler.loadTests();
    },

    debug: true
  };
};
```

### Code editor or IDE name and version

WebStorm 2018.1?

### OS name and version

OSX
10.13.4

### Old config that worked with v5 of Angular 

This is from my actual project. 

```js
var wallabyWebpack = require("wallaby-webpack");
var path = require("path");

var compilerOptions = Object.assign(
  require("./tsconfig.json").compilerOptions,
  require("./src/tsconfig.spec.json").compilerOptions
);

compilerOptions.module = "CommonJs";

module.exports = function(wallaby) {
  var webpackPostprocessor = wallabyWebpack({
    entryPatterns: ["src/wallabyTest.js", "src/**/*spec.js"],

    module: {
      rules: [
        { test: /\.css$/, loader: ["raw-loader", "css-loader"] },
        { test: /\.html$/, loader: "raw-loader" },
        {
          test: /\.ts$/,
          loader: "@ngtools/webpack",
          include: /node_modules/,
          query: { tsConfigPath: "tsconfig.json" }
        },
        {
          test: /\.js$/,
          loader: "angular2-template-loader",
          exclude: /node_modules/
        },
        { test: /\.json$/, loader: "json-loader" },
        { test: /\.styl$/, loaders: ["raw-loader", "stylus-loader"] },
        { test: /\.less$/, loaders: ["raw-loader", "less-loader"] },
        {
          test: /\.scss$|\.sass$/,
          loaders: [
            "raw-loader",
            {
              loader: "sass-loader",
              options: {
                includePaths: [
                  path.join(wallaby.projectCacheDir, "src/style-vars")
                ]
              }
            }
          ]
        },
        { test: /\.(jpg|png)$/, loader: "url-loader?limit=128000" }
      ]
    },

    resolve: {
      extensions: [".js", ".ts"],
      modules: [
        path.join(__dirname, "typings"),
        path.join(wallaby.projectCacheDir, "src/app"),
        path.join(wallaby.projectCacheDir, "src"),
        "node_modules",
        path.join(__dirname, "../node_modules")
      ]
    },
    node: {
      fs: "empty",
      net: "empty",
      tls: "empty",
      dns: "empty"
    }
  });

  return {
    files: [
      {
        pattern: "src/**/*.+(ts|css|less|scss|sass|styl|html|json|svg)",
        load: false
      },
      { pattern: "typings/**/*", load: true },
      { pattern: "src/**/*.d.ts", ignore: true },
      { pattern: "src/**/*spec.ts", ignore: true }
    ],

    tests: [
      { pattern: "src/**/*spec.ts", load: false },
      { pattern: "src/**/*e2e-spec.ts", ignore: true }
    ],

    testFramework: "jasmine",

    compilers: {
      "**/*.ts": wallaby.compilers.typeScript(compilerOptions)
    },

    middleware: function(app, express) {
      var path = require("path");
      app.use(
        "/favicon.ico",
        express.static(path.join(__dirname, "src/favicon.ico"))
      );
      app.use("/assets", express.static(path.join(__dirname, "src/assets")));
    },

    env: {
      kind: "electron"
    },

    postprocessor: webpackPostprocessor,

    setup: function() {
      window.__moduleBundler.loadTests();
    },

    debug: true
  };
};
```

**Comment by ArtemGovorov** [comment]
In your basic example repo `ng test` produces the same error. My guess is that because the `typings` folder is missing. 

Anyway, the issue is not Angular version specific, and we have the docs section explaining [how to map TypeScript paths to webpack settings](https://wallabyjs.com/docs/integration/typescript.html#typescript-compileroptionspaths-setting). You were close to get it working, but webpack alias format is different from Typescript paths, and the [`/*` postfix is not recognised by webpack](https://webpack.js.org/configuration/resolve/).

You don't need any `typings` folder in `resolve.modules`, just need a correct alias (without `*` and ideally pointing either to wallaby cache dir):

```js
"@gtypes": path.join(wallaby.projectCacheDir, 'typings')
```

and `typings/**/*` in your `files` list (ideally with `load: false`), just like you have it in your old config.

I have [sent you the PR](https://github.com/vespertilian/ngCliWebpackSample/pull/1).

I'm also not sure how it could work in Angular 5, you should have needed the same alias in wallaby config for Angular 5 as well.

**Comment by vespertilian** [comment]
@ArtemGovorov 

Thank you for the quick reply, even on a Saturday. Sorry all the files were not there, I didn't notice that .gitignore had the typings directory ignored for the angular cli project when I committed my changes. 

With your help, I got it working, I updated the repo if you want to see. 

I only had to add the  `{pattern: "typings/**/*", load: true }`

I did not need the alias or the modules path

```javascript
    files: [
      {pattern: 'src/**/*.+(ts|css|less|scss|sass|styl|html|json|svg)', load: false},
      {pattern: 'src/**/*.d.ts', ignore: true},
      {pattern: 'src/**/*spec.ts', ignore: true},
      {pattern: "typings/**/*", load: true }
    ],
```
